codechain_primitives/UnsignedInteger.py

```diff
@@ -0,0 +1,140 @@
+from rlp import encode
+
+
+class _UnsignedInteger(int):
+    """UnsignedInteger base class: all unsigned integer inherit from this class.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if type(args[0]) is str:
+            result = super().__new__(cls, *args, **kwargs, base=0)
+        else:
+            result = super().__new__(cls, *args, **kwargs)
+
+        if result < 0:
+            raise ValueError("Integer underflow")
+        if hasattr(cls, 'MAX_VALUE') and result > cls.MAX_VALUE:
+            raise ValueError("Integer overflow")
+        return result
+
+    def plus(self, rhsValue):

```

Comment:
Do we need this method form arithmetic operations when we have operator overriding?

Reply:
Okay, I removed it.